### PR TITLE
feat(sunburst): use arcs package

### PR DESCRIPTION
**Pull request by plouc**
This PR makes use of the new `@nivo/arcs` package:
- this way we can remove some components such as labels
- we have more efficient transitions which are now handled globally instead of using `useSpring` for every element
- we also have enter/exit transitions (can be seen in action in the drill down story :)) 
- we now have a `transitionMode` which can be used to control the way nodes enter/exit
- some types can be used directly from this package such as the arc labels properties, which are now identical with `@nivo/pie`
- the arc label component can optionally use a custom component and the position relative to the radius can be configured via `arcLabelsRadiusOffset`

I've also revised the types to use generic types from both `@nivo/core` and `@nivo/arcs`, also changed the `ComputedDatum` to be more aligned with other packages.
I've added a `path` property to the nodes, this way we can easily get a list of all the ancestors of a node, I thought it could be useful.

There was a bug with the data generator for this chart, if you look at the current website, you'll see that the generated data get nested under the root `name` property each time you click `roll the dice`.

I realized that the pie package no more require `lodash` (at least not as a direct dependency), and same for `react-spring`, so I removed them.

There's also another change completely unrelated on the references page.

### CODE REVIEW

**Review by wyze — CHANGES_REQUESTED** [review]

File: packages/sunburst/tests/Sunburst.test.tsx [hunk 1]
```diff
@@ -195,76 +195,81 @@ describe('Sunburst', () => {
...
-            expect(slices.at(4).prop('node').data.id).toEqual('A-1-I')
-            expect(slices.at(4).prop('node').data.color).toEqual('#ff5500')
+            expect(arcs.at(4).prop('datum').id).toEqual('A-1-I')
+            expect(arcs.at(4).prop('datum').color).toEqual('#ff5500')
         })
     })
 
     describe('patterns & gradients', () => {
-        it('should support patterns', () => {
+        xit('should support patterns', () => {
```
[the remark this answers is not in the document]
Comment (wyze) [reply packages/sunburst/tests/Sunburst.test.tsx:272]: It seems like the changes to `bindDefs` cause this to fail. What is the reason behind those changes?

File: packages/sunburst/src/types.ts [hunk 2]
```diff
@@ -111,18 +98,25 @@ export type SvgProps<RawDatum> = DataProps<RawDatum> &
...
     Partial<CommonProps<RawDatum>>
 
+export type CompleteSvgProps<RawDatum> = DataProps<RawDatum> &
+    Dimensions &
+    SvgDefsAndFill<RawDatum> &
+    MouseEventHandlers<RawDatum, SVGPathElement> &
+    ModernMotionProps &
+    CommonProps<RawDatum>
+
 export type SunburstArcProps<RawDatum> = Pick<
```
[the remark this answers is not in the document]
Comment (plouc) [reply packages/sunburst/src/types.ts]: Good catch! Removed.

File: packages/sunburst/src/types.ts [hunk 3]
```diff
@@ -111,18 +98,25 @@ export type SvgProps<RawDatum> = DataProps<RawDatum> &
...
     SvgProps<RawDatum>,
     'onClick' | 'onMouseEnter' | 'onMouseLeave' | 'onMouseMove' | 'borderWidth' | 'borderColor'
 > &
     Pick<CommonProps<RawDatum>, 'isInteractive' | 'tooltip'> & {
-        arcGenerator: Arc<any, ComputedDatum<RawDatum>>
+        arcGenerator: ArcGenerator
         node: ComputedDatum<RawDatum>
     }
 
 export type SunburstLabelProps<RawDatum> = {
```
[the remark this answers is not in the document]
Comment (plouc) [reply packages/sunburst/src/types.ts]: Good catch! Removed.

**Review by wyze — APPROVED** [review]

**Comment by TheMatrixMaster** [comment]
When will this PR be merged?

**Review by TheMatrixMaster — APPROVED** [review]

**Pull request merged**